fritz

```diff
@@ -128,145 +185,85 @@ def check_config(cfg="fritz.defaults.yaml", yes=False, **kwargs):
+    return response
 
-    init_file = "skyportal/skyportal/__init__.py"
-    with open(init_file, "r") as f:
-        init = f.readlines()
-    out = []
-    for line in init:
-        if line.startswith("__version__ = "):
-            __version__ = "+".join(
-                [v for v in __version__.split("+") if not v.startswith("fritz")]
-            )
-            line = f'__version__ = "{__version__}+fritz.{git_hash}"\n'
-        out.append(line)
-    with open(init_file, "wb") as f:
-        f.write("".join(out).encode("utf-8"))
 
-    # get the git log for SP and add it to the SP distro
-    p = subprocess.run(
+def get_skyportal_token():
+    """Get admin token for Kowalski to access SkyPortal"""
+    result = subprocess.run(
         [
-            "git",
-            "--git-dir=.git",
-            "log",
-            "--no-merges",
-            "--pretty=format:[%cI %h %cE] %s",
-            "-1000",
+            "docker",
+            "exec",
+            "-i",
+            "skyportal_web_1",
+            "/bin/bash",
+            "-c",
+            "cat /skyportal/.tokens.yaml",
         ],
+        cwd="skyportal",
         capture_output=True,
         universal_newlines=True,
-        cwd="skyportal",
     )
-    out = p.stdout
-    with open("skyportal/data/gitlog.txt", "w") as spgl:
-        spgl.write(out)
-
-    # add Fritz-specific dependencies for SP
-    # js
-    with open("extensions/skyportal/package.fritz.json", "r") as f:
-        fritz_pkg = json.load(f)
-    with open("skyportal/package.json", "r") as f:
-        skyportal_pkg = json.load(f)
-
-    skyportal_pkg["dependencies"] = {
-        **skyportal_pkg["dependencies"],
-        **fritz_pkg["dependencies"],
-    }
-    with open("skyportal/package.json", "w") as f:
-        json.dump(skyportal_pkg, f, indent=2)
+    token = result.stdout.split()[-1]
 
-    # python
-    with open(".requirements/ext.txt", "r") as f:
-        ext_req = f.readlines()
-    with open("skyportal/requirements.txt", "r") as f:
-        skyportal_req = f.readlines()
-    with open("skyportal/requirements.txt", "w") as f:
-        f.writelines(skyportal_req)
-        for line in ext_req:
-            if line not in skyportal_req:
-                f.write(line)
+    if len(token) != 36:
+        raise RuntimeError("Failed to get a SkyPortal token for Kowalski")
 
-    # The skyportal .dockerignore includes config.yaml, but we would like
-    # to add that file; we therefore overwrite the .dockerignore
-    with open("skyportal/.dockerignore", "w") as f:
-        f.write(".*\n")
+    return token
 
 
-def build(args):
-    """
-    Build Fritz
+def build(
+    init: bool = False,
+    repo: str = "origin",
+    branch: str = "master",
+    traefik: bool = False,
+    no_kowalski: bool = False,
+    do_update: bool = False,
+    skyportal_tag: str = "skyportal/web:latest",
+    yes: bool = False,
+):
+    """Build Fritz
+
+    :param init: Initialize before updating Fritz
```

Comment:
This seems different from everywhere else

Reply:
Fixed the description.